app/controllers/work_packages/moves_controller.rb

```diff
@@ -33,80 +33,44 @@ class WorkPackages::MovesController < ApplicationController
   before_filter :authorize
 
   def new
-    prepare_for_work_package_move
+    WorkPackageBulkUpdateService.new(@work_packages, @projects)
+      .run(params).available_attributes.each do |key, value|
+        instance_variable_set(:"@#{key}", value)
+      end
     render layout: false if request.xhr?
   end
 
   def create
-    prepare_for_work_package_move
-
-    new_type = params[:new_type_id].blank? ? nil : @target_project.types.find_by_id(params[:new_type_id])
-    unsaved_work_package_ids = []
-    moved_work_packages = []
-    @work_packages.each do |work_package|
-      work_package.reload
-
-      call_hook(:controller_work_packages_move_before_save,  params: params, work_package: work_package, target_project: @target_project, copy: !!@copy)
-
-      permitted_params = params.permit(:copy,
-                                       :assigned_to_id,
-                                       :responsible_id,
-                                       :start_date,
-                                       :due_date,
-                                       :priority_id,
-                                       :follow,
-                                       :new_type_id,
-                                       :new_project_id,
-                                       ids: [],
-                                       status_id: [])
-
-      if r = work_package.move_to_project(@target_project, new_type,  copy: @copy,
-                                                                      attributes: permitted_params,
-                                                                      journal_note: @notes)
-        moved_work_packages << r
-      else
-        unsaved_work_package_ids << work_package.id
+    WorkPackageBulkUpdateService.new(@work_packages, @projects)
+      .save(params, true).each do |key, value|
+        instance_variable_set(:"@#{key}", value)
       end
-    end
-    set_flash_from_bulk_work_package_save(@work_packages, unsaved_work_package_ids)
+
+    set_flash_from_bulk_work_package_save(@work_packages, @unsaved_work_packages)
 
     if params[:follow]
-      if @work_packages.size == 1 && moved_work_packages.size == 1
-        redirect_to work_package_path(moved_work_packages.first)
+      if @work_packages.size == 1 && @moved_work_packages.size == 1
+        redirect_to work_package_path(@moved_work_packages.first)
       else
         redirect_to project_work_packages_path(@target_project || @project)
       end
     else
       redirect_to project_work_packages_path(@project)
     end
-      end
+  end
 
-  def set_flash_from_bulk_work_package_save(work_packages, unsaved_work_package_ids)
-    if unsaved_work_package_ids.empty? and not work_packages.empty?
+  def set_flash_from_bulk_work_package_save(work_packages, unsaved_work_packages)
```

Comment:
Assignment Branch Condition size for set_flash_from_bulk_work_package_save is too high. [15.3/15]